upm-1776 private networking (#1855)
EnterpriseDB/docs: product_docs/docs/edbcloud/beta/using_cluster/connecting_your_cluster/02_connect_on_premises.mdx

---
title: Connecting from Networks Outside Azure
---

If your applications are deployed on your on-premises environments, or on other Cloud providers instead Azure, you can set up a secure tunnel between your on premises network and the Azure virtual network. 

> There are a few options you can use.

Reviewer: We only describe one option - let's cut this and simplify. "EDB recommends using Point-to-Site Azure VPN Gateways" or something.

## Connecting Using Azure VPN Gateway (Point-to-Site)

> Azure VPN gateway provides you an encrypted tunnel to communicate

Reviewer: This is a weird paragraph - can we delete it? It sounds like an ad for VPN Gateway. If we want to keep it, let's rewrite it to be more direct about the EDB Cloud specific benefits. The next paragraph already does that, though...

with your IT infrastructures on a Azure Private Network over complex
network environments like the internet.

Point-to-Site VPN connections are useful when you want to connect to
your EDB Cloud cluster from a remote location, such as when you do

> administrative operations from home. You can also use P2S instead of a

Reviewer: ```suggestion
administrative operations from home. You can also use Point-to-Site instead of a
```

Reviewer: This change needs to be made in several places in the doc.

> Site-to-Site VPN when you have only a few clients that need to connect

Reviewer: We should talk about the benefits of P2S here, not just gesture towards the limitations. /cc @zozowell

to your EDB Cloud service. Point-to-Site connections do not require a VPN
device or a public-facing IP address. P2S creates the VPN connection

> over either SSTP (Secure Socket Tunneling Protocol) or IKEv2 protocol.

Reviewer: ```suggestion
over either SSTP (Secure Socket Tunneling Protocol) or IKEv2.
```

The Azure resources that are created when you create a virtual network

> gateway include:

Reviewer: Are there more than these? Curious about the word "include".

> -   A VM that is deployed in your EDB Cloud cluster virtual network. It works as a VPN application to handle all traffic.

Reviewer: ```suggestion
-   A VM that is deployed in your EDB Cloud cluster virtual network. This VM works as a VPN application to handle all traffic network traffic.
```

-   A dedicated subnet called `GatewaySubnet` to accommodate the VPN VM.

## Point-to-Site Example

These are the steps to connect using the Azure VPN Gateway Point-to-Site option. 

### Step 1: Create Virtual Network VPN Gateway

>  In the **Virtual network gateway** page, create the first VPN gateway for the EDB Cloud Cluster resource virtual network `vnet-japaneast`. Name the VPN gateway "vpng-edbcloud."

Reviewer: I think "Virtual Network Gateway" is a proper noun. But I have no idea what page this is referring to.

>  In the **Virtual network gateway** page, create the first VPN gateway for the EDB Cloud Cluster resource virtual network `vnet-japaneast`. Name the VPN gateway "vpng-edbcloud."

Reviewer: > Name the VPN gateway "vpng-edbcloud."

This won't work if there are multiple VPN gateways, right @zozowell?

>  In the **Virtual network gateway** page, create the first VPN gateway for the EDB Cloud Cluster resource virtual network `vnet-japaneast`. Name the VPN gateway "vpng-edbcloud."

Reviewer: > virtual network `vnet-japaneast`

We should explain that this is an example, and they might have a different VNet name, just like other examples.

 ![](images/image8.png)

!!! Note

>     The VPN gateway automatically creates a dedicated subnet to accommodate its gateway VMs. Ensure that your EDB Cloud virtual network address space has sufficient IP range for the subnet to prevent errors in the virtual network.

Reviewer: How can the customer actually do this, @zozowell?

> ### Step 2: Create self-signed root CA for your VPN gateway authentication

Reviewer: ```suggestion
### Step 2: Create self-signed root CA for your VPN Gateway authentication
```

Certificates are used by Azure to authenticate clients connecting to a
VNet over a Point-to-Site VPN connection. See
[Generate a Self-Signed Root
Certificate](https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-howto-point-to-site-resource-manager-portal#getcer).

> This example uses a self-signed certificate, but EDB recommends you use your Enterprise certificate in

Reviewer: > you use your Enterprise certificate

@zozowell: Is this something that Azure customers should understand? I'm not sure I follow.

production.

> ### Step 3: Point-to-site configuration in EDB Cloud VPN gateway

Reviewer: This isn't an EDB Cloud VPN Gateway.

Reviewer 2: How about call it "Postgresql Cluster VPN gateway"?

> ### Step 3: Point-to-site configuration in EDB Cloud VPN gateway

Reviewer: ```suggestion
### Step 3: Point-to-Site configuration in EDB Cloud VPN Gateway
```

Configure a point-to-site (P2S) connection for VPN gateway `vpng-edbcloud` from the Azure virtual network
gateway portal. See [Create the VPN gateway](https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-howto-point-to-site-resource-manager-portal#creategw).

![Point-to-site configuration](images//point-to-site.png)

> -   **Address pool** - the reserved private IP address pool for every client connected through VPN gateway. It should be in your virtual network IP range, but not overlap with subnets in use.

Reviewer: > It should be in your virtual network IP range, but not overlap with subnets in use.

Isn't this the EDB Cloud subnet range? How are they supposed to do this?

Reviewer 2: It is! thanks for pointing this out...  
We have abundant ip address in AKS vnet, I think it's isolated dedicated vpn subnet. I need to confirm.

---

> title: Connecting from an On-Premises Network

Reviewer: I think we should call this "Connecting from Networks Outside Azure".

---

Sometimes your applications are deployed on your on-premises
environments, or on other Cloud providers instead Azure, but they also

> want to consume private EDB Cloud services from these premises via a

Reviewer: I'm not sure the applications "want" anything 😁

> secure approach. In these cases, you can set up a secure tunnel

Reviewer: I think we need to talk about motivations here. This is only marginally more secure, and the customer is jumping through a lot of hoops when they can just use the public endpoints that we make available.

@zozowell: Would you be okay with not having this page in the docs at all?

Reviewer 3: @ba-edb , which do you prefer:

- Not having a this page in the docs at all
- Keep the page along with the motivations mentioned by Zane as follws:
 
 If your applications are deployed on your on-premises environments, or on other Cloud providers instead Azure, you can set up a secure tunnel between your on premises network and the Azure virtual network. Setting up a secure tunnel enables you to connect remotely or from your organization's staging or production environment.

Reviewer: Alright, let's keep it. I'll take another review pass through it.
